extension/src/datum_utils.rs

```diff
@@ -560,12 +575,8 @@ impl<'a> IntoIterator for DatumStore<'a> {
             if (*tentry).typbyval {
                 // Datum by value
                 DatumStoreIntoIterator::Value {
-                    // SAFETY `data` is guaranteed to be 8-byte aligned, so it should be safe to use as a slice
-                    iter: std::slice::from_raw_parts(
-                        self.data.as_slice().as_ptr() as *const Datum,
-                        self.data_len as usize / 8,
-                    )
-                    .iter(),
```

Comment:
Wow, look at that! We returned a reference to self and then let it go out of scope and dropped! And now there's a test case for it! :)

I think there's still room for improvement here, though perhaps you want to wait for a follow-up. Not only do we have unsafe operations mixed in and a way over-scoped unsafe block, but we manage an iterator by hand, which just adds a source of bugs. We can both see that your iterator is correct now (can we? I'm pretty sure... but this code already had one bug...), but will it always remain so as the system evolves?

I put the changes I have in mind into commit https://github.com/timescale/timescaledb-toolkit/commit/d1783b3db23f343267c1e756624fef4ee3d8ae7e - let me know what you think.

Also, while I don't see any similar bug in DatumStore, it would certainly benefit from similar improvements. Another good candidate for a follow-up.

Thanks!